app/scripts/lib/multichain-api/MultichainMiddlewareManager.test.ts

```diff
@@ -0,0 +1,70 @@
+import { JsonRpcRequest } from '@metamask/utils';
+// eslint-disable-next-line import/no-named-as-default
+import createMultichainMiddlewareManager, {
+  ExtendedJsonRpcMiddleware,
+} from './MultichainMiddlewareManager';
+
+describe('MultichainMiddlewareManager', () => {
+  it('should add middleware and get called for the scope', () => {
+    const multichainMiddlewareManager = createMultichainMiddlewareManager();
+    const middlewareSpy = jest.fn() as unknown as ExtendedJsonRpcMiddleware;
+    (middlewareSpy as { destroy: () => void }).destroy = jest.fn();
+    const domain = 'example.com';
+    multichainMiddlewareManager.addMiddleware(
+      'eip155:1',
+      domain,
+      middlewareSpy,
+    );
+    multichainMiddlewareManager.middleware(
+      { scope: 'eip155:1' } as unknown as JsonRpcRequest,
+      { jsonrpc: '2.0', id: 0 },
+      () => {
+        //
+      },
+      () => {
+        //
+      },
+    );
+    expect(middlewareSpy).toHaveBeenCalled();
+  });
+  it('should remove middleware', () => {
+    const multichainMiddlewareManager = createMultichainMiddlewareManager();
+    const middlewareMock = jest.fn() as unknown as ExtendedJsonRpcMiddleware;
+    middlewareMock.destroy = jest.fn();
```

Comment:
should we be checking that this is called?

Reply:
Is it called? Why are we setting it to a jest.fn() at all?